### PR TITLE
feat: configure protected topics via cli

### PR DESCRIPTION
**Pull request by alrevuelta**
Summary:
* Allows to protect custom pusub topics with a given public key, provided via cli arguments.
* New flag `protected-topic` is added with the format `pubsubtopic:publickey`
* Continuation of https://github.com/waku-org/nwaku/pull/1614
* Related https://github.com/waku-org/nwaku/issues/1612

Example:

```
./build/wakunode2 \
--topics="/waku/2/topic-1/proto /waku/2/topic-2/proto"\
--protected-topic=/waku/2/topic-1/proto:0425b97b092867bf74532d96e50b6227450b1413078efd414929ff444ece0f9146e58d6ac785c13e76111841812d6ddd20c83c88a6a24815948896731410b0c70b \
--protected-topic=/waku/2/topic-2/proto:04e790854444ae882cc69b25ec7bd454e7bdb8d4b900a4add3eeb52cf62415d9d93a5c7eb3c9ea08e8e854d8b8ac26f5073aaac6fa68c69b8f078bbd48c2d5589d
```

Output:

```
NOT 2023-04-19 16:39:53.554+02:00 routing only signed traffic                topics="wakunode" tid=3718069 file=wakunode2.nim:452 protectedTopic=/waku/2/topic-2/proto publicKey=04e790854444ae882cc69b25ec7bd454e7bdb8d4b900a4add3eeb52cf62415d9d93a5c7eb3c9ea08e8e854d8b8ac26f5073aaac6fa68c69b8f078bbd48c2d5589d
NOT 2023-04-19 16:39:53.554+02:00 routing only signed traffic                topics="wakunode" tid=3718069 file=wakunode2.nim:452 protectedTopic=/waku/2/topic-1/proto publicKey=0425b97b092867bf74532d96e50b6227450b1413078efd414929ff444ece0f9146e58d6ac785c13e76111841812d6ddd20c83c88a6a24815948896731410b0c70b
```

- [x] Blocked by wakunode2 changes, wait before merging.
- [x] Move config to `Application-level configuration`

### CODE REVIEW

**Comment by LNSD** [comment]
Would you mind waiting for wakunode2 feature branch to be merged to merge this PR? cc @jm-clius

**Review by jm-clius — COMMENTED** [review]
In general, LGTM. We should be able to merge the `wakunode2` app changes early next week, after which this should be mergeable. Mind moving the config item to a separate section with a commented title such as `Application-level configuration`. This is because it configures only application behaviour and not strictly speaking the node.

File: apps/wakunode2/config.nim [hunk 1]
```diff
@@ -457,6 +466,19 @@ proc parseCmdArg*(T: type crypto.PrivateKey, p: string): T =
...
   return @[]
 
+proc parseCmdArg*(T: type ProtectedTopic, p: string): T =
+  let elements = p.split(":")
+  if elements.len != 2:
+    raise newException(ConfigurationError, "Invalid format for protected topic expected topic:publickey")
+
+  let publicKey = secp256k1.SkPublicKey.fromHex(elements[1])
+  if publicKey.isErr:
+    raise newException(ConfigurationError, "Invalid public key")
```
Comment (Ivansete-status) [inline apps/wakunode2/config.nim:476]: Shall we append the `publicKey.error()`at this point?

File: apps/wakunode2/config.nim [hunk 2]
```diff
@@ -533,6 +555,17 @@ proc readValue*(r: var EnvvarReader, value: var crypto.PrivateKey) {.raises: [Se
   except CatchableError:
     raise newException(SerializationError, getCurrentExceptionMsg())
 
+proc readValue*(r: var TomlReader, value: var ProtectedTopic) {.raises: [SerializationError].} =
+  try:
+    value = parseCmdArg(ProtectedTopic, r.readValue(string))
+  except CatchableError:
+    raise newException(SerializationError, getCurrentExceptionMsg())
```
Comment (Ivansete-status) [inline apps/wakunode2/config.nim:562]: Maybe we could reduce some code:
```suggestion
proc readValue*(r: var TomlReader, value: var ProtectedTopic) {.raises: [ConfigurationError].} =
    value = parseCmdArg(ProtectedTopic, r.readValue(string))
```

Comment (alrevuelta) [reply apps/wakunode2/config.nim:563]: mm unsure how this works under the hood, so just mirroring what we currently have in other `readValue` function parsing other arguments.

**Comment by alrevuelta** [comment]
@jm-clius with the `wakunode2` changes merged, this should be ready to go. Move also the config to a "appication-level" seciton.

**Review by jm-clius — APPROVED** [review]
LGTM, thanks!

**Pull request merged**